rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,621 @@
+sharing sessions.
+
+## Details
+
+After signing into a Team plan cluster, which currently has no connected nodes, the user sees a
+prompt in the server resource table encouraging them to use CMC (screenshot 1 below). Interacting
+with the CTA opens a new tab which explains what happens during the setup and lets the user start
+the setup (screenshot 2 below).
+
+![CTA](assets/0133-cta.png)
+
+![setup landing](assets/0133-setup-landing.png)
+
+The setup requires no extra configuration. It sets up relevant roles, reissues the user cert to
+refresh the roles, downloads the teleport binary, creates a join token, calls `teleport node
+configure`, and then launches the agent by calling `teleport start`.
+
+![setup progress](assets/0133-setup-progress.png)
+
+
+The setup steps are displayed in a similar fashion to Cloud account setup steps when the cluster is
+being spun up for the first time. Once the setup is done, the setup tab automatically transitions to
+showing the status of the agent.
+
+Outside of the setup, the user is able to manage CMC through a dedicated top bar UI and a status
+tab, similar to the gateway tab. The top bar UI allows the user to quickly start or stop the agent
+and to open the status tab. The status tab displays errors related to starting or stopping the
+agent. It also lets the user remove the agent completely.
+
+### Initial setup
+
+Let’s assume that we have a Team plan cluster at acme.teleport.sh with a user robert@acme.com.
+robert@acme.com uses bob as the system username on their computer.
+
+Once the user begins the setup, Connect performs all steps without further user interaction. Each
+step can independently fail with an error message explaining what happened. The setup process can be
+retried on error.
+
+#### Setting up roles
+
+
+The setup starts with the creation of the relevant role. If the user does not have permissions to
+create and set roles, this step fails with an error message explaining which permissions are needed.
+
+The role provides the necessary system login and access privileges. This way we have a single role
+providing both rather than a role which provides just the login and then CMC having to depend on
+another role, such as the built-in access, to provide access privileges.
+
+The role is scoped per cluster user. Its name follows the format of
+`connect-my-computer-<cluster-username>`. This is to accommodate multiple cluster users with the
+same system username wanting to use CMC – their CMC roles will not conflict with each other.
+
+For our example user, the YAML representation of the role looks like this:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: connect-my-computer-robert@acme.com
+spec:
+  allow:
+  node_labels:
+    teleport.dev/connect-my-computer/owner: robert@acme.com
+  logins: [bob]
+```
+
+Connect creates this role and adds it to the current user. Since the user cert in Connect at the
```

Comment:
@zmb3 @xinding33 @klizhentas Could we limit Connect My Computer to only local users? This likely won't impact our target audience much, as initially Team plan clusters consist solely of local users until SSO is set up.

The context is that Nic pointed out that adding roles to SSO users isn't straightforward, as their roles are dictated by the auth connector setup. An SSO user is an ephemeral resource on the backend, so we cannot just add a role to it.

We need to create a role in the first place to allow users to log into the Connect My Computer node with their system user. For example, robert@acme.com should be able to log in as bob.

I looked at the SSO docs and I don't see a way around it, at least not without significant effort past what we have originally planned. We'd likely need to treat Connect My Computer as a first-class citizen on the backend, while so far we wanted to create the MVP by using existing cluster features, without pushing changes to the backend.

Reply:
Yeah, let's just limit to local users for now. For our hobbyist/home lab use case it will almost always be the first local user who signed up, especially if we get the onboarding/invite flow right.

Reply:
Yes, we will enforce it by only showing the option for local users.

IP pinning won't be necessary here.